Dev

TimofeyKhach opened the pull request.

TimofeyKhach committed:
added models and schemas

TimofeyKhach committed:
restructuring

TimofeyKhach committed:
models one style

TimofeyKhach committed:
chto-to

TimofeyKhach committed:
paths beginning

l8beOne committed:
Add base routers

l8beOne committed:
Add base routers 1

TimofeyKhach committed:
added user create schemas

TimofeyKhach committed:
Merge branch 'dev' of github.com:l8beOne/automation_bot into dev

TimofeyKhach committed:
added user routers

mephistiks committed:
Update: module structure, crud, base. Fix: circular import

l8beOne committed:
Add bot app(fix file structure), add some of the keyboards and handlers

l8beOne committed:
Add buttons and keyboard with information of HSE

TimofeyKhach committed:
some new routers created

TimofeyKhach committed:
Merge branch 'dev' of github.com:l8beOne/automation_bot into dev

TimofeyKhach committed:
added session for GET requests

mephistiks reviewed (commented):
1. Меняем в зависимостях генартор на try версие вместо with
2. попробовать удалить foreignkey из моделей и прокинуть его напрямую в Join запрос
3. в декораторах роутеров aiogram кучу сравнений поменять на in и сравнивать с массивом вариантов
4. raise httpexception вызывать строго в роутерах, ни в коем случае не в crud
5. разделить роутеры в app/schedule/router.py на отдельные файлы/директории

mephistiks reviewed (changes requested):
1. Меняем в зависимостях генартор на try версие вместо with
2. попробовать удалить foreignkey из моделей и прокинуть его напрямую в Join запрос
3. в декораторах роутеров aiogram кучу сравнений поменять на in и сравнивать с массивом вариантов
4. raise httpexception вызывать строго в роутерах, ни в коем случае не в crud
5. разделить роутеры в app/schedule/router.py на отдельные файлы/директории

l8beOne committed:
Fix filters, make keyboards sync, put out texts of keyboards in another file

l8beOne committed:
Merge branch 'dev' of github.com:l8beOne/automation_bot into dev

mephistiks commented on bot/handlers/contacts.py:
```diff
@@ -1,53 +1,35 @@
 
-
-@router.message(F.text.lower() == "контакты")
+@router.message(F.text.lower().in_({"контакты"}))
```
Если уж и оптимизировать, то можно было использовать frozenset а не set

mephistiks reviewed (approved):
Если мёрдж конфликта нет, то сливайте, теперь прям отличный код

TimofeyKhach committed:
db port problem fixed
added new func for db sessions: get_db()
routers separeted
added crud for registration with group
added cruds for schedules, programs, subjects, groups

TimofeyKhach committed:
db port problem fixed,
> added new func for db sessions: get_db(),
> routers separeted,
> added crud for registration with group,
> added cruds for schedules, programs, subjects, groups

TimofeyKhach committed:
Merge branch 'dev' of github.com:l8beOne/automation_bot into dev

mephistiks commented on app/routers/groups.py:
```diff
@@ -0,0 +1,32 @@
+    db: AsyncSession = Depends(get_db)
+):
+    result = await create_group_crud(new_group, session, db)
+    if type(result) == dict and result.get("status") == "error":
```
если здесь будет возвращаться НЕ dict, питон всё равно попытается сделать result.get и прокинет ошибку

mephistiks commented on app/schedule/crud.py:
```diff
@@ -80,32 +118,110 @@ async def create_group_crud(
+            "status": "error",
+            "status_code": 400,
+            "detail": "Group already registered"
+        }
```
зачем создавать целый объект ради ответа, можно же просто вернуть tuple, либо если хотите структурами, так создайте какой-то объект означающий ответ, то есть MyResponse

mephistiks commented on app/routers/subjects.py:
```diff
@@ -0,0 +1,28 @@
+
+from ..database import get_db
+from ..schedule.crud import create_subject_crud, get_subjects_crud
+from ..schedule.schemas import SubjectBaseSchema, SubjectGetSchema
```
вынесите то что используете везде в глобальный crud.py

mephistiks commented on docker-compose.yml:
```diff
@@ -4,10 +4,10 @@ services:
     ports:
-      - 5434:5432
+      - 5430:5432
     environment:
```
прокидывайте такие вещи через .env

https://docs.docker.com/compose/environment-variables/set-environment-variables/

mephistiks commented on docker-compose.yml:
```diff
@@ -4,10 +4,10 @@ services:
     image: postgres
     ports:
-      - 5434:5432
+      - 5430:5432
```
прокидывайте такие вещи через .env

https://docs.docker.com/compose/environment-variables/set-environment-variables/

mephistiks commented on docker-compose.yml:
```diff
@@ -4,10 +4,10 @@ services:
     environment:
       - POSTGRES_HOST=localhost
-      - POSTGRES_PORT=5434
+      - POSTGRES_PORT=5430
```
Ахахахаха, я сейчас объясню что вы сделали, вот
```text
Docker      | PC
5430(sql)  -> None(его никто не ждёт тут)
5432(None) -> 5430
```
При этом всём вы в своих конфигах подключаетесь вообще к порту 5434

mephistiks reviewed (changes requested):
Исправить всё что в комментариях
Глобальные функции(те которые используются в нескольких местах) вынести глобально
Прибраться в файлах конфигурации и докере, а то вообще не понятно что вы там куда хотите прокинуть

mephistiks commented:
Плюс у вас всё ещё конфликтов несколько, благо в файлах в которых всё легко чинится!

l8beOne committed:
Make structure of announe sender

l8beOne committed:
Merge branch 'dev' of github.com:l8beOne/automation_bot into dev

l8beOne committed:
Fix router.py, r.in, r.txt; add announce sender

chetaleks committed:
fix: better schedule output

TimofeyKhach committed:
change function "create_subject_crud" (fix problem with adding several subjects)
added function "get_specific_schedule_crud"
change interaction with user (using telegram_id instead of id)

TimofeyKhach committed:
Merge branch 'dev' of github.com:l8beOne/automation_bot into dev

l8beOne committed:
The implementation of the mailing list has been fully completed

l8beOne committed:
merge with branch dev

mephistiks commented:
ПОЧИНИТЕ МЁРДЖ КОНФЛИКТ!!!!

l8beOne committed:
Add AdminCommandFilter, fix announcement sender
